### PR TITLE
PXC-4316: Nodes "changing identity" can prevents primary groups

### PR DESCRIPTION
**Pull request by kamil-holubicki**
https://perconadev.atlassian.net/browse/PXC-4316

Problem:
If the node is partitioned from the cluster, and during this time it shuts down and starts again and then rejoins the cluster, the other part of the cluster still waits for partitioned node.

Cause:
Node's uuid is maintained in gvwstate.dat file. If the node shuts down gracefully, the file is deleted. When the node starts and there is no gvwstate.dat file, new uuid is assigned to this node. When such a node rejoins the cluster, it is identified by new uuid. gmcast layer properly detects that the node changed its identity, but this information is not propagated to upper protocol layers. pc layer maintains its own set of known nodes. As the information about identity changing is not delivered to this layer, it sees the "new" node identified by new uuid, but also waits for "old" node identified by old uuid.

Solution:
Propagate information about node's identity change to upper protocol layers.

### CODE REVIEW

**Comment by kamil-holubicki** [comment]
https://pxc.cd.percona.com/view/8.0%20parallel%20MTR/job/pxc-8.0-pipeline-parallel-mtr/2330/

File: gcomm/src/pc_proto.hpp [hunk 1]
```diff
@@ -244,6 +244,7 @@ class gcomm::pc::Proto : public Protolay
     void handle_user(const Message&, const Datagram&,
                      const ProtoUpMeta&);
     void deliver_view(bool bootstrap = false);
+    void identity_changed(const gcomm::UUID& old_identity, const gcomm::UUID& new_identity);
```
Comment (venkatesh-prasad-v) [inline gcomm/src/pc_proto.hpp]: The below version is more readable
```suggestion
    void identity_changed(const gcomm::UUID& old_identity, 
                          const gcomm::UUID& new_identity);
```

File: gcomm/src/view.cpp [hunk 2]
```diff
@@ -57,6 +57,7 @@ static std::string to_string(const gcomm::ViewType type)
     case gcomm::V_REG:      return "REG";
     case gcomm::V_NON_PRIM: return "NON_PRIM";
     case gcomm::V_PRIM:     return "PRIM";
+    case gcomm::V_IDENTITY_CHANGE: return "V_IDENTITY_CHANGE";
```
Comment (venkatesh-prasad-v) [inline gcomm/src/view.cpp]:
```suggestion
    case gcomm::V_IDENTITY_CHANGE: return "IDENTITY_CHANGE";
```

File: gcomm/src/pc_proto.hpp (hunk 1 above)
Comment (kamil-holubicki) [reply gcomm/src/pc_proto.hpp]: done

File: gcomm/src/view.cpp (hunk 2 above)
Comment (kamil-holubicki) [reply gcomm/src/view.cpp]: done

**Review by venkatesh-prasad-v — APPROVED** [review]
LGTM

**Pull request merged**